scripts/mc_launch_rtp_daemon.py

```diff
@@ -0,0 +1,200 @@
+        if len(existing_files) == 0:
+            warnings.warn("None of these files are on disk. Flagging as failed.")
+            continue
+
+        # scan files if desired
+        obsids = []
+        if SCAN_FILES:
+            print("Scanning files")
+            from pyuvdata import UVData
+
+            bad_metadata_files = []
+            for filename in filelist:
+                try:
+                    uvd = UVData()
+                    uvd.read(filename, read_data=False)
+                except (KeyError, OSError, ValueError):
+                    bad_metadata_files.append(filename)
+                    filelist.remove(filename)
+                    if RENAME_BAD_FILES:
+                        os.rename(filename, filename + BAD_SUFFIX)
+                else:
+                    # if file is valid, add obsid to running list
+                    starttime = Time(
+                        np.unique(uvd.time_array)[0], scale="utc", format="jd"
+                    )
+                    obsids.append(int(np.floor(starttime.gps)))
+
+        # go to working directory
+        os.chdir(WORKING_DIRECTORY)
+        jd_folder = os.path.join(os.getcwd(), f"{jd:d}")
+        if not os.path.isdir(jd_folder):
+            os.makedirs(jd_folder)
+        os.chdir(jd_folder)
+
+        # copy config file as a reference
+        shutil.copy2(WORKFLOW_CONFIG, jd_folder)
+
+        # make name of output makeflow file
+        mf_filename = os.path.basename(WORKFLOW_CONFIG).rstrip(".toml") + ".mf"
+        mf_filename = os.path.join(jd_folder, mf_filename)
+
+        print("Building Makeflow")
+        # make a workflow
+        mt.build_makeflow_from_config(filelist, WORKFLOW_CONFIG, mf_filename, jd_folder)
+
+        # launch workflow inside of tmux
+        cmd = (
+            "source /home/obs/.bashrc; "
+            "conda deactivate; "
+            f"conda activate {CONDA_ENV}; "
+            f"makeflow -T slurm {mf_filename} -J 100; /bin/bash -l"
+        )
+        session_name = f"rtp_{jd}"
+        tmux_cmd = ["tmux", "new-session", "-d", "-s", session_name, cmd]
+        try:
+            print("Launching:", tmux_cmd)
+            subprocess.check_call(tmux_cmd)
+        except subprocess.CalledProcessError as e:
+            warnings.warn(
+                f"Error spawning tmux session: command was {e.cmd}; "
+                f"returncode was {e.returncode:d}; output was {e.output}; "
+                f"stderr was {e.stderr}."
+            )
+            continue
+
+        # if we get to here, assume launch was successful and delete jd from
+        # list of unprocessed
+        print(f"Removing redis key JD={jd}")
+        r.hdel(JD_KEY, jd_redis)
+
+        # update RTP launch records
+        print("Updating RTP launch records")
+        if len(obsids) == 0:
+            obsids = _get_obsids(filelist)
+        t0 = Time.now()
+        with db.sessionmaker() as session:
+            for filename, obsid in zip(filelist, obsids):
```

Comment:
It looks like `filename` is not used in the body of this loop, so we probably only need to iterate over `obsids`.

Reply:
done.